samples/hssi/hssi_10g_cmd.h

```diff
@@ -252,6 +267,96 @@ class hssi_10g_cmd : public hssi_cmd
+
+    std::cout << "HSSI performance: " << std::endl;
+
+    res = fpgaGetProperties(NULL, &pprops);
+    if (res != FPGA_OK) {
+    	std::cerr << "Failed to get properties" << std::endl;
+    	return;
+    }
+
+    res = fpgaPropertiesSetObjectType(pprops, FPGA_DEVICE);
+    if (res != FPGA_OK) {
+    	std::cerr << "Failed to set type to FPGA_DEVICE" << std::endl;
+    	return;
+    }
+
+    fpgaEnumerate(&pprops, 1, &token, 1, &num_matches);
+
+    if (0 != num_matches) {
+    	// Retrieve the Bitstream ID 
+    	fpgaDestroyProperties(&pprops);
+    	fpgaGetProperties(token, &pprops);
+    	res = fpgaPropertiesGetBBSID(pprops, &bitstream_id);
+    	if (res != FPGA_OK) {
+    		std::cerr << "Failed to read bitstream id from properties" << std::endl;
+    		return;
+    	}
+
+	major = FPGA_BBS_VER_MAJOR(bitstream_id);
+	if (major == BITSVER_MAJOR_S10)
+            clk_freq = USER_CLKFREQ_S10;
+        else if (major == BITSVER_MAJOR_AC)
+            clk_freq = USER_CLKFREQ_AC;
+	else {
+	        fpgaDestroyToken(&token);
+		fpgaDestroyProperties(&pprops);
+		std::cout << "unknown platform" << std::endl;
+		return;
+	}
+
+    	fpgaDestroyToken(&token);
+    }
+    else
+    {
+    	std::cerr << "No FPGA found" << std::endl;
+        fpgaDestroyProperties(&pprops);
+        return;
+    }
+
+    fpgaDestroyProperties(&pprops);
+
+    // Reading traffic control Tx/Rx timestamp registers 
+    uint32_t tx_end_tstamp = hafu->mbox_read(CSR_TX_END_TSTAMP);
+    uint32_t rx_sta_tstamp = hafu->mbox_read(CSR_RX_STA_TSTAMP);
+    uint32_t rx_end_tstamp = hafu->mbox_read(CSR_RX_END_TSTAMP);
+
+    // Convert timestamp register from clock cycles to nanosecond
+    double sample_period_ns = 1000 / clk_freq;
+    double tx_end_tstamp_ns = tx_end_tstamp * sample_period_ns;
+    double rx_sta_tstamp_ns = rx_sta_tstamp * sample_period_ns;
+    double rx_end_tstamp_ns = rx_end_tstamp * sample_period_ns;
+    
+    // Calculate latencies
+    double latency_min_ns = rx_sta_tstamp_ns; 
+    double latency_max_ns = rx_end_tstamp_ns - tx_end_tstamp_ns;
+
+    // Calculate Tx/Rx throughput achieved
+    uint32_t data_pkt_size = packet_length_ - 4;
+    double total_tx_duration_ns = tx_end_tstamp_ns;
+    double total_rx_duration_ns = (rx_end_tstamp_ns - rx_sta_tstamp_ns);
+    uint32_t total_data_size_bits = (num_packets_  * data_pkt_size * 8) ;
+    double achieved_tx_tput_gbps = (total_data_size_bits / total_tx_duration_ns);
+    double achieved_rx_tput_gbps = (total_data_size_bits / total_rx_duration_ns);
+
+    std::cout << "\tSelected clock frequency : "<< clk_freq << " MHz" << std::endl
+    	      << "\tLatency minimum : " << latency_min_ns <<" ns"<<std::endl
+    	      << "\tLatency maximum : " << latency_max_ns <<" ns"<<std::endl
+    	      << "\tAchieved Tx throughput : " << achieved_tx_tput_gbps << " GB/s" << std::endl
+    	      << "\tAchieved Rx throughput : " << achieved_rx_tput_gbps << " GB/s" << std::endl
+    	      << std::endl;
```

Comment:
The remainder of the hssi application is written with the C++ API.  Why have we gone back to C here?